### PR TITLE
Re-implement label ignore feature.

### PR DESCRIPTION
**Pull request by benpatterson**
This reverts commit 6fe5895d283c8a70d245d86813626040112c0995.

This re-implements on #431.

CC @udisch

### CODE REVIEW

File: src/main/java/org/jenkinsci/plugins/ghprb/GhprbPullRequest.java [hunk 1]
```diff
@@ -165,9 +167,29 @@ public void check(GHPullRequest ghpr, boolean isWebhook) {
...
         checkSkipBuild();
+        checkLabels();
         tryBuild();
     }
 
+    private void checkLabels() {
+        Set<String> labelsToIgnore = helper.getLabelsIgnoreList();
+        if (labelsToIgnore != null && !labelsToIgnore.isEmpty()) {
+            try {
+                for (GHLabel label : pr.getLabels()) {
```
Comment (benpatterson) [inline src/main/java/org/jenkinsci/plugins/ghprb/GhprbPullRequest.java]: `pr.getLabels()` is the call that ends up re-writing the pr object with an incorrect url. We'll need to fix this area of code or benefit from an upstream fix.

File: README.md [hunk 2]
```diff
@@ -51,7 +51,8 @@ For more details, see https://wiki.jenkins-ci.org/display/JENKINS/GitHub+pull+re
   * The phrase for adding users to the whitelist via comment. (Java regexp)  
   * The phrase for accepting a pull request for testing. (Java regexp)
   * The phrase for starting a new build. (Java regexp)  
-  * The crontab line. This specify default setting for new jobs.  
+  * The crontab line. This specify default setting for new jobs.
```
Comment (mmitche) [inline README.md]: specifies

File: src/main/java/org/jenkinsci/plugins/ghprb/GhprbPullRequest.java (hunk 1 above)
Comment (mmitche) [inline src/main/java/org/jenkinsci/plugins/ghprb/GhprbPullRequest.java]: Do you know whether this makes a GitHub API call?

File: src/main/java/org/jenkinsci/plugins/ghprb/jobdsl/GhprbContextExtensionPoint.java [hunk 3]
```diff
@@ -39,6 +39,7 @@ public Object githubPullRequest(Runnable closure) throws ANTLRException {
                 null,
                 null,
                 null,
+                null,
```
Comment (mmitche) [inline src/main/java/org/jenkinsci/plugins/ghprb/jobdsl/GhprbContextExtensionPoint.java]: Can you add a context field for this so it is configurable through DSL?

**Comment by benpatterson** [comment]
Folks - The 'dirty' request that the underlying api library uses has been [fixed upstream](https://github.com/kohsuke/github-api/commit/3a66e90b7aadbd43c70a2ae391c39af037ae0415).

**Comment by udisch** [comment]
@benpatterson Can you elaborate what's missing to make this work? Also some thoughts after the previous issue:
1) Is there a place to add an integration test with `github-api`?
2) Do you know if the plugin always uses the correct version of `github-api`, or it might use an existing version if the jenkins installation already has one.

Thanks

**Comment by udisch** [comment]
Thanks again for everyone working on this and getting it merged.
I noticed that if you add a label that's on the whitelist, the change will only get picked up when closing and re-opening the pull request. In my original PR, commenting "jenkins retest this please" was enough. It seems the issue details are cached? I guess the way around it would be to add an implementation which uses the web hooks.

**Pull request merged**